src/GraphQL.Tests/Utilities/SchemaBuilderTests.cs

```diff
@@ -432,6 +467,87 @@ Stars description
             input.Fields.First().Description.ShouldBe("Stars description");
         }
 
+        [Fact]
+        public void builds_input_types_with_default_values()
+        {
+            var definitions = @"
+                input ReviewInput {
+                  stars: Int! = 23
+                }
+            ";
+
+            var schema = Schema.For(definitions);
+            schema.Initialize();
+
+            var input = schema.AllTypes["ReviewInput"] as InputObjectGraphType;
+            input.ShouldNotBeNull();
+            input.Fields.Count.ShouldBe(1);
+            input.Fields.First().DefaultValue.ShouldBe(23);
+        }
+
+        [Fact]
+        public void builds_input_types_with_dependent_default_values()
+        {
+            var definitions = @"
+                input SomeInputType1 {
+                  test: SomeInputType2! = { arg1: 22 }
+                }
+                input SomeInputType2 {
+                  arg1: Int
+                  arg2: Int = 30
+                }
+            ";
+
+            var schema = Schema.For(definitions);
+            schema.Initialize();
+
+            var inputType1 = schema.AllTypes["SomeInputType1"] as InputObjectGraphType;
+            inputType1.ShouldNotBeNull();
+            inputType1.Fields.Count.ShouldBe(1);
+            inputType1.Fields.First().Name.ShouldBe("test");
+            var value1 = inputType1.Fields.First().DefaultValue.ShouldBeAssignableTo<IDictionary<string, object>>();
+            value1.ShouldContainKeyAndValue("arg1", 22);
+            value1.ShouldContainKeyAndValue("arg2", 30);
+
+            var inputType2 = schema.AllTypes["SomeInputType2"] as InputObjectGraphType;
+            inputType2.ShouldNotBeNull();
+            inputType2.Fields.Count.ShouldBe(2);
+            inputType2.Fields.Find("arg1").ShouldNotBeNull().DefaultValue.ShouldBeNull();
+            inputType2.Fields.Find("arg2").ShouldNotBeNull().DefaultValue.ShouldBe(30);
+        }
+
+        [Fact]
+        public void input_types_default_value_loops_throw1()
```

Comment:
Did you see https://github.com/graphql/graphql-spec/pull/793#issuecomment-726811961 ?

Worth to asd test like 
```graphql
input O {
  o: O = { o: null }
  a: Int
}
```

Reply:
Also I see in their discussion those defaults : `= { }` . What does that mean ?

Reply:
The example you stated is not currently supported.  At present, it will throw a friendly error message stating that circular dependencies are not supported.

Supporting circular dependencies (when possible) is outside of my goal for this PR.  Rather, the goal is to prevent out-of-order dependencies from crashing the GraphQL.NET infrastructure.  This PR solves that.  You can add an issue or PR to support circular dependencies.

Reply:
> Also I see in their discussion those defaults : `= { }` . What does that mean ?

It means an object of the specified type with default values.  This is supported by GraphQL.NET.

Reply:
> Did you see [graphql/graphql-spec#793 (comment)](https://github.com/graphql/graphql-spec/pull/793#issuecomment-726811961) ?

I started to thumb through it.  It seems they are dealing with the spec, trying to decide if the spec should explicitly describe coercion rules.  But coercion must take place within GraphQL.NET whether it's in the spec or not, or GraphQL.NET will not run correctly.  So I don't think it really applies to us, unless they decide to change a coercion rule (for example, if they decide that circular coercion should not be allowed).

Reply:
If and when they merge a change to the spec, I'll review at that time.  Regardless, this PR is necessary for GraphQL.NET to operate correctly.

Reply:
What about adding test for 
```graphql
input O {
  o: O = { o: null }
  a: Int
}
```
even if it fails for now. Mark as ignored with comment pointing this thread.

Reply:
sure

Reply:
done